File: prueba4-back/src/main/java/com/getfit/servicios/PlanService.java
package com.getfit.servicios;

import java.util.Set;

import com.getfit.modelo.Entrenador;
import com.getfit.modelo.Plan;

public interface PlanService {

	Set<Plan> obtenerPlanes();

	Plan agregarPlan(Plan plan);

	Plan actualizarPlan(Plan plan);

	Plan obtenerPlan(Long planId);

	void eliminarPlan(Long planId);

	Plan listarPlan(Long planId);

}

File: GetFit/tfg-master/prueba4-front/src/app/services/plan.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class PlanService {

  constructor(private http:HttpClient) { }

  public guardarPlan(plan:any){
    return this.http.post(`${baserUrl}/plan/`,plan);
  }

  public eliminarPlan(planId:any){
    return this.http.delete(`${baserUrl}/plan/${planId}`);
  }

  public actualizarPlan(plan:any){
    return this.http.put(`${baserUrl}/plan/`,plan);
  }

  public obtenerPlan(planId:any){
    return this.http.get(`${baserUrl}/plan/${planId}`);
  }

  public listarPlanes(){
    return this.http.get(`${baserUrl}/plan/`);
  }

}